util/include/openssl_utils.hpp

```diff
@@ -0,0 +1,83 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+#pragma once
+
+#include <string>
+#include <memory>
+#include <fstream>
+
+#include <openssl/bio.h>
+#include <openssl/ec.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+#include <openssl/evp.h>
+
+#include "crypto_utils.hpp"
+
+using std::pair;
+using std::make_pair;
+using std::string;
+using std::unique_ptr;
+using std::ofstream;
+using concord::util::crypto::KeyFormat;
+
+namespace concord::crypto::openssl {
+
+class CertificateUtils {
+ public:
+  static std::string generateSelfSignedCert(const std::string& origin_cert_path,
+                                            const std::string& pub_key,
+                                            const std::string& signing_key);
+  static bool verifyCertificate(X509* cert, const std::string& pub_key);
```

Comment:
why are you using raw pointers and not references?

Reply:
These are existing implementations. I had just moved from one file to another.
Though, the changes from raw pointers to references are done.

Reply:
They appeared as new, which is why I asked.
If you didn't implement this, It might be wrong to change it, as I assume you didn't check every call site for its type.
Every caller which passes a char* to a function which receives a const std::string& creates a copy of the string.